### ISSUE TITLE
Distributing Android library thats using NDK, cant use native methods?

### ISSUE THREAD

**Issue by Orbyt** [jitpack/jitpack.io#617]
I'm still pretty unfamiliar with the NDK, so keep that in mind. I am trying to distribute an Android library that uses the ndk. The library is the popular BASS audio library: http://www.un4seen.com/

I have the repo on github and can look up the repository on jitpack.io and am able to add it to my project. I added the proper `compile 'com.github.rockmyworld:basslib-android:1.6'` line to my build.gradle, and i know its hooked up properly because I can get autocomplete suggestions for a test class I put in the library.

That said, using any of the actual native methods throws a `java.lang.UnsatisfiedLinkError`: No implementation found for...

Why is this? AFAIK, jitpack does support ndk builds as noted in this issue: https://github.com/jitpack/jitpack.io/issues/24

So any ideas?

**Comment by jitpack-io** [comment]
Yes, NDK is supported and it looks like the compilation was successful. The resulting .aar file contains some native .so libraries.
One approach to debugging this would be to build the library locally on your machine and then try to use it.

**Comment by Orbyt** [comment]
Found the solution. The package name has to comply with some stuff in the ndk part, in this case:

`com.un4seen.bass`.

It was changed by another team member internally, hence why it previously worked.

Thanks for the swift response.

**Comment by ballenjr** [comment]
Hello @Orbyt ,

Could you be more specific about where this package name is important? I am having trouble with OpenCV static files being added to a project. They are added and similarly the build log is successful.

But there is a weird comment in the log that mentions missing source files pertaining to JNI. But the really weird thing is that all of the .so files that ended up in the apk are around 5KB. And they are renamed to the project name. Which threw me for a loop. 

Which is, also, not the result that I get when I add them using cordova or a plain/non-modular/not-pluggable Android Studio project. It works fine under these conditions.

I am wondering if the renaming of the files has anything to do with what you are talking about? 
Also I am not sure if the gradle setup is missing anything to facilitate the JNI stuff? Anyone know if this is a consideration?

Here is the log: https://jitpack.io/com/github/ballenjr/OpenCV-3.1.0-Android/-85f2d61146-1/build.log

Perhaps if I apply the correct convention in the correct context the planets will align to let this work...?

**Comment by ballenjr** [comment]
UPDATE: I finally found success. Not sure that it had anything to do with a package naming convention though. The last thing that I tried before it worked was reducing the number of architectures that I included. Then it made it in. I did make a change to the overall project name implicitly. But that didn't have any noticeable effect until I made the last change and attempt. And the name change didn't relate to anything anywhere else in the project. So I am still unsure what you mean.

OPENSOURCEORDIE
